### PR TITLE
Check email query

### PR DESCRIPTION
**Pull request by Angham116**
#21 made ```checkEmail``` query with a test for all expected results.

### CODE REVIEW

**Review by ahmedalmi95 — APPROVED** [review]
white space

File: src/database/config/db_build.sql [hunk 1]
```diff
@@ -42,4 +42,7 @@ CREATE TABLE requirement (
 	prop_id INT REFERENCES proposal(id)
 );
 
-COMMIT ;
+INSERT INTO users(firstname, lastname, email, freelancer_url, photo_url, mobile_number, password)
+VALUES ('a', 'b', 'a.gmail.com', 'https://github.com/angham', 'https://www.iconspng.com/image/36709/face-avatar-man-male-handsome-3', '0097234567', '1234');
+
+COMMIT
```
Comment (Fatmasiam) [inline src/database/config/db_build.sql]: add new line

File: src/database/config/db_build.sql [hunk 2]
```diff
@@ -42,4 +42,7 @@ CREATE TABLE requirement (
 	prop_id INT REFERENCES proposal(id)
 );
 
-COMMIT ;
+INSERT INTO users(firstname, lastname, email, freelancer_url, photo_url, mobile_number, password)
+VALUES ('a', 'b', 'a.gmail.com', 'https://github.com/angham', 'https://www.iconspng.com/image/36709/face-avatar-man-male-handsome-3', '0097234567', '1234');
```
Comment (tomduggan85) [inline src/database/config/db_build.sql]: @Angham116 You could use the hashed and salted version of 1234 as your password value that you store in the database here, which would let you type 1234 into the login field and successfully log in with this account.

File: test/test.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+      return checkEmail(email);
+    })
+    .then((result) => {
+      t.equal(result.rows[0].email, 'a.gmail.com', 'the first row should be not empty');
+      t.end();
+    })
+    .catch((error) => {
+      t.error(error);
+    });
+});
```
Comment (tomduggan85) [inline test/test.js:34]: @Angham116 Nice job adding tests!

Comment (Angham116) [reply test/test.js:34]: thanks

**Review by Anies12 — APPROVED** [review]

File: package.json [hunk 4]
```diff
@@ -31,6 +31,8 @@
     "eslint": "^5.15.3",
     "eslint-config-airbnb-base": "^13.1.0",
     "eslint-plugin-import": "^2.16.0",
-    "nodemon": "^1.18.10"
+    "nodemon": "^1.18.10",
```
Comment (kannanhassouna) [inline package.json:36]: ``` nodemon ``` ?

File: src/database/config/db_build.sql [hunk 5]
```diff
@@ -59,3 +59,5 @@ INSERT INTO requirement(text,prop_id)
 values ('full time' ,1), ('react',1);
 
 COMMIT ;
+COMMIT;
```
Comment (kannanhassouna) [inline src/database/config/db_build.sql:61]: ``` commit ``` twice  ?

File: test/test.js [hunk 6]
```diff
@@ -1,6 +1,45 @@
...
+    .then((result) => {
+      t.equal(result.rows[0], undefined, 'the first row should be empty');
+      t.end();
+    })
+    .catch((error) => {
+      t.error(error);
+    });
+});
+
+tape('Test checkEmail query function if there is no email match with income email', (t) => {
```
Comment (kannanhassouna) [inline test/test.js:23]: you can make your tests cleaner 

```javascript
 tape('Test checkEmail query function if there is email match with income email', (t) => {
  reBuildDB()
    .then(() => checkEmail('a@gmail.com'))
    .then((result) => {
      t.equal(result.rows[0], undefined, 'should not there any values in rows');
      t.end();
    })
    .catch(error =>  {
       t.error(error)
       t.end();
     });
});
```

**Review by kannanhassouna — APPROVED** [review]

File: test/test.js [hunk 7]
```diff
@@ -1,6 +1,39 @@
...
+      t.error(error);
+      t.end();
+    });
+});
+
+tape('Test checkEmail query function if there is no email match with income email', (t) => {
+  reBuildDB()
+    .then(() => checkEmail('a.gmail.com'))
+    .then((result) => {
+      t.equal(result.rows[0].email, 'a.gmail.com', 'the first row should be not empty');
```
Comment (kannanhassouna) [inline test/test.js]:
```suggestion
      t.equal(result.rows[0].email, undefined, 'the first row should be not empty');
```

**Review by kannanhassouna — CHANGES_REQUESTED** [review]

File: src/database/config/db_build.sql (hunk 5 above)
Comment (Fatmasiam) [reply src/database/config/db_build.sql:61]: @knanahassouna1  just one ?!

File: test/test.js (hunk 6 above)
Comment (Fatmasiam) [reply test/test.js:23]: @Angham116  as the last test !!!

**Pull request closed without merging**